packages/protocol/contracts/identity/FederatedAttestations.sol

```diff
@@ -412,58 +411,41 @@ contract FederatedAttestations is
-        // attestations before they have been unauthorized?
-        require(
-          account == msg.sender || getAccounts().attestationSignerToAccount(msg.sender) == issuer,
-          "Sender does not have permission to revoke this attestation"
-        );
-        // This is meant to delete the attestation in the array
-        // and then move the last element in the array to that empty spot,
-        // to avoid having empty elements in the array
-        // TODO benchmark gas cost saving to check if array is of length 1
-        identifierToAttestations[identifier][issuer][i] = attestations[attestations.length - 1];
-        identifierToAttestations[identifier][issuer].pop();
-
-        bool deletedIdentifier = false;
-        bytes32[] memory identifiers = addressToIdentifiers[account][issuer];
-        for (uint256 j = 0; j < identifiers.length; j = j.add(1)) {
-          if (identifiers[j] == identifier) {
-            addressToIdentifiers[account][issuer][j] = identifiers[identifiers.length - 1];
-            addressToIdentifiers[account][issuer].pop();
-            deletedIdentifier = true;
-            break;
-          }
-        }
-        // Should never be false - both mappings should always be updated in unison
-        assert(deletedIdentifier);
-
-        bytes32 attestationHash = getUniqueAttestationHash(
-          identifier,
-          issuer,
-          account,
-          signer,
-          issuedOn
-        );
-        // Should never be able to re-revoke an attestation
-        assert(!revokedAttestations[attestationHash]);
-        revokedAttestations[attestationHash] = true;
-
-        emit AttestationRevoked(identifier, issuer, account, signer, issuedOn, publishedOn);
-        return;
-      }
+    require(
+      account == msg.sender ||
+        // Minor gas optimization to prevent storage lookup in Accounts.sol if issuer == msg.sender
+        issuer == msg.sender ||
+        getAccounts().attestationSignerToAccount(msg.sender) == issuer,
+      "Sender does not have permission to revoke this attestation"
+    );
+    _revokeAttestation(identifier, issuer, account);
+  }
+
+  /**
+   * @notice Revokes attestations [identifiers <-> accounts] from issuer
+   * @param issuer Address of the issuer of all attestations to be revoked
+   * @param identifiers Hash of the identifiers
+   * @param accounts Addresses of the accounts mapped to the identifiers
+   *   at the same indices
+   * @dev Throws if the number of identifiers and accounts is not the same
+   * @dev Throws if sender is not the issuer or currently registered signer of issuer
+   * @dev Throws if an attestation is not found for identifiers[i] <-> accounts[i]
+   */
+  function batchRevokeAttestations(
+    address issuer,
+    bytes32[] calldata identifiers,
+    address[] calldata accounts
+  ) external {
+    require(identifiers.length == accounts.length, "Unequal number of identifiers and accounts");
```

Comment:
do we want to enforce a limit here for the max length of attestations that can be passed in?

Reply:
I was thinking we would do that in the SDK, but I'll add a reviewer TODO so we can get more feedback on that!